Include/cpython/unicodeobject.h

```diff
@@ -280,26 +341,24 @@ PyAPI_FUNC(int) _PyUnicode_CheckConsistency(
 #define SSTATE_INTERNED_IMMORTAL 2
 
 /* Use only if you know it's a string */
-#define PyUnicode_CHECK_INTERNED(op) \
-    (((PyASCIIObject *)(op))->state.interned)
+static inline int PyUnicode_CHECK_INTERNED(PyObject *op) {
```

Comment:
to avoid the cast going away, consider doing what Py_INCREF did & add indirection through a macro for the cast:

```
#define PyUnicode_CHECK_INTERNED(op) \
    _PyUnicode_CHECK_INTERNED(_PyASCIIObject_CAST(op))
```

Reply:
The SC asked to not add such macro :-) You're right that without such macro, there is a risk of introducing new compiler warnings.

If possible I would prefer to keep ``PyObject*`` for functions in unicodeobject.c, since it's the type used for arguments in existing functions and the type returned by functions creating strings like PyUnicode_New(), PyUnicode_FromString(), etc.

Maybe for this specific header file, we can avoid casts.

For me, PyASCIIObject is an implementation detail which should be hidden. If possible, it should even be moved to the internal C API, but that's a way larger topic which may require a PEP ;-)